packages/module/src/FieldBuilder/FieldBuilder.tsx

```diff
@@ -0,0 +1,330 @@
+  Grid,
+  GridItem,
+} from '@patternfly/react-core';
+import { PlusCircleIcon, MinusCircleIcon } from '@patternfly/react-icons';
+
+/**
+ * Defines the helpers passed to the children render prop.
+ * This allows each rendered row to have access to its own specific index.
+ */
+export interface FieldRowHelpers {
+  rowIndex: number;
+  /** 
+   * Ref callback to attach to the first focusable element in the row.
+   * This enables automatic focus management when rows are added/removed.
+   */
+  focusRef: (element: HTMLElement | null) => void;
+  /**
+   * Unique ID for this row group - use this for aria-labelledby associations
+   */
+  rowGroupId: string;
+  /**
+   * ID for the first column label - use this to associate the first field with its column header
+   */
+  firstColumnLabelId: string;
+  /**
+   * ID for the second column label - use this to associate the second field with its column header
+   */
+  secondColumnLabelId?: string;
+  /**
+   * ID for the row label - use this in combination with column labels for comprehensive labeling
+   */
+  rowLabelId: string;
+  /**
+   * Complete aria-labelledby string for the first column that includes both row and column context
+   */
+  firstColumnAriaLabelledBy: string;
+  /**
+   * Complete aria-labelledby string for the second column that includes both row and column context
+   */
+  secondColumnAriaLabelledBy?: string;
+}
+
+/**
+ * Extends FormGroupProps to inherit standard functionality like
+ * label, helperText, isRequired, and validation states.
+ */
+export interface FieldBuilderProps extends Omit<FormGroupProps, 'children'> {
+  /** Label for the first column (required for both single and two-column layouts) */
+  firstColumnLabel: React.ReactNode;
+  /** Label for the second column (optional, only used in two-column layout) */
+  secondColumnLabel?: React.ReactNode;
+  /** The total number of rows to render. This should be derived from the length of the state array managed by the parent. */
+  rowCount: number;
+  /**
+   * A function that returns the content for each row. This "render prop" provides
+   * maximum flexibility for defining the inputs within each row.
+   * Can return 1 child (single-column) or 2 children (two-column).
+   */
+  children: (helpers: FieldRowHelpers) => React.ReactNode;
+  /** A callback triggered when the "Add" button is clicked. */
+  onAddRow: () => void;
+  /** A callback triggered when a "Remove" button is clicked, which receives the index of the row to remove. */
+  onRemoveRow: (index: number) => void;
+  /** Optional props to customize the "Add" button. */
+  addButtonProps?: Omit<ButtonProps, 'onClick'>;
+  /**
+   * Optional label prefix for each row group. Defaults to "Row".
+   * Screen readers will announce this as "Row 1", "Row 2", etc.
+   */
+  rowGroupLabelPrefix?: string;
+  /**
+   * Optional unique ID prefix for this FieldBuilder instance.
+   * This ensures unique IDs when multiple FieldBuilders exist on the same page.
+   */
```

Comment:
We can omit "optional" in the prop description since that will be conveyed by the props table.

```suggestion
   * Unique ID prefix for this FieldBuilder instance. Each FieldBuilder must have a unique ID prefix passed into it.
   */
```